provider/rancher/api.go

```diff
@@ -197,35 +194,35 @@ func listRancherContainer(client *rancher.RancherClient) []*rancher.Container {
 	return containerList
 }
 
-func parseAPISourcedRancherData(environments []*rancher.Environment, services []*rancher.Service, containers []*rancher.Container) []rancherData {
+func parseAPISourcedRancherData(stacks []*rancher.Stack, services []*rancher.Service, containers []*rancher.Container) []rancherData {
 	var rancherDataList []rancherData
 
-	for _, environment := range environments {
+	for _, stack := range stacks {
 
 		for _, service := range services {
 
-			if service.EnvironmentId != environment.Id {
+			if service.StackId != stack.Id {
 				continue
 			}
 
 			rancherData := rancherData{
-				Name:       environment.Name + "/" + service.Name,
+				Name:       service.Name + "/" + stack.Name,
 				Health:     service.HealthState,
 				State:      service.State,
 				Labels:     make(map[string]string),
 				Containers: []string{},
 			}
 
 			if service.LaunchConfig == nil || service.LaunchConfig.Labels == nil {
-				log.Warnf("Rancher Service Labels are missing. Environment: %s, service: %s", environment.Name, service.Name)
+				log.Warnf("Rancher Service Labels are missing. Stack: %s, service: %s", stack.Name, service.Name)
 			} else {
 				for key, value := range service.LaunchConfig.Labels {
 					rancherData.Labels[key] = value.(string)
 				}
 			}
 
 			for _, container := range containers {
-				if container.Labels[labelRancherStackServiceName] == rancherData.Name &&
+				if container.Labels[labelRancherStackServiceName] == stack.Name+"/"+service.Name &&
```

Comment:
Because one thing is how system establish context hierarchy names for services ```stack/service```  set by container label ```io.rancher.stack_service.name```

Another thing is the dns name for the services```service.stack.domain``` to set default traefik Host matcher.. All rancher dns external integrations and internal dns set by default dns entries as ```service.stack.domain``` . I think it's useful to make them match by default.